FileShareClient/FileShareApiClient.cs

```diff
@@ -106,6 +112,50 @@ public async Task<Stream> DownloadFileAsync(string batchId, string filename)
             }
         }
 
+        public async Task<IResult<DownloadFileResponse>> DownloadFileAsync(string batchId, string fileName, string downldLocation, long fileSizeInBytes = 0, CancellationToken cancellationToken = default)
+        {
+            long startByte = 0;
+            long endByte = fileSizeInBytes < maxDownloadBytes ? fileSizeInBytes - 1 : maxDownloadBytes;
+            var result = new Result<DownloadFileResponse>();
+            HttpStatusCode httpStatusCode = HttpStatusCode.OK;
+
+            while (startByte <= endByte)
+            {
+                string rangeHeader = $"bytes={startByte}-{endByte}";
+
+                var uri = $"batch/{batchId}/files/{fileName}";
+
+                using (var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, uri))
+                {
+                    var httpClient = await GetAuthenticationHeaderSetClient();
+                    if (fileSizeInBytes != 0 && rangeHeader != null)
+                    {
+                        httpRequestMessage.Headers.Add("Range", rangeHeader);
+                        httpStatusCode = HttpStatusCode.PartialContent;
+                    }
+
+                    var response = await httpClient.SendAsync(httpRequestMessage, CancellationToken.None);
```

Comment:
Please use the `cancellationToken` parameter here instead of `CancellationToken.None`

Reply:
update the same. Thanks